x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/join/Join.java

```diff
@@ -236,4 +236,10 @@ public void postAnalysisVerification(Failures failures) {
             }
```

Comment:
This PR is not about validation, but above we specifically disallow the `text` type, and I think we do nothing for `semantic_text`. I feel like that likely needs to be excluded, too.

Reply:
Semantic text is covered here, at least according to the PR at https://github.com/elastic/elasticsearch/pull/125581.